qiskit/transpiler/passes/layout/csp_layout.py

```diff
@@ -97,48 +97,64 @@ def __init__(self, coupling_map, strict_direction=False, seed=None, call_limit=1
         self.strict_direction = strict_direction
         self.call_limit = call_limit
         self.time_limit = time_limit
-        self.seed = seed
+        self.seed = seed or 42
 
-    def run(self, dag):
-        qubits = dag.qubits
-        cxs = set()
+        random.seed(self.seed)
 
-        for gate in dag.two_qubit_ops():
-            cxs.add((qubits.index(gate.qargs[0]),
-                     qubits.index(gate.qargs[1])))
-        edges = set(self.coupling_map.get_edges())
+    def run(self, dag):
+        """ run the layouting method """
+        logical_qubits = dag.qubits
 
         if self.time_limit is None and self.call_limit is None:
-            solver = RecursiveBacktrackingSolver()
-        else:
-            solver = CustomSolver(call_limit=self.call_limit, time_limit=self.time_limit)
-
-        problem = Problem(solver)
-        problem.addVariables(list(range(len(qubits))), self.coupling_map.physical_qubits)
-        problem.addConstraint(AllDifferentConstraint())  # each wire is map to a single qubit
-
-        if self.strict_direction:
-            def constraint(control, target):
-                return (control, target) in edges
+            csp_solver = RecursiveBacktrackingSolver()
         else:
-            def constraint(control, target):
-                return (control, target) in edges or (target, control) in edges
-
-        for pair in cxs:
-            problem.addConstraint(constraint, [pair[0], pair[1]])
+            csp_solver = CustomSolver(call_limit=self.call_limit, time_limit=self.time_limit)
 
-        random.seed(self.seed)
+        problem = self._get_constraint_satisfaction_problem(dag, self.coupling_map, csp_solver)
         solution = problem.getSolution()
 
         if solution is None:
             stop_reason = 'nonexistent solution'
-            if isinstance(solver, CustomSolver):
-                if solver.time_current is not None and solver.time_current >= self.time_limit:
+            if isinstance(csp_solver, CustomSolver):
+                if csp_solver.time_current is not None and csp_solver.time_current >= self.time_limit:
                     stop_reason = 'time limit reached'
-                elif solver.call_current is not None and solver.call_current >= self.call_limit:
+                elif csp_solver.call_current is not None and csp_solver.call_current >= self.call_limit:
                     stop_reason = 'call limit reached'
         else:
             stop_reason = 'solution found'
-            self.property_set['layout'] = Layout({v: qubits[k] for k, v in solution.items()})
+            self.property_set['layout'] = Layout({v: logical_qubits[k] for k, v in solution.items()})
 
         self.property_set['CSPLayout_stop_reason'] = stop_reason
+
+    def _get_constraint_satisfaction_problem(self, dag, coupling_map, csp_solver):
```

Comment:
Is this refactor necessary? it makes the patch hard to read...

Reply:
From my point of view it increases readability and allows one to understand the inner mechanics of CSPLayout. Nevertheless, I removed it as requested.

Reply:
Thanks! I'm okey with that refactor happening in a following PR, it just make this one a bit dirty...